1-Draft/RFCxxxx-PSModulePath.md

```diff
@@ -0,0 +1,159 @@
+|---------------------|--------------------------------------------------------------------------|---------------------------------------------------|
+| User modules        | C:\Users\slee\Documents\WindowsPowerShell\Modules                        | C:\Users\slee\Documents\PowerShell\Modules        |
+| System modules      | C:\Program Files\WindowsPowerShell\Modules                               | C:\Program Files\PowerShell\Modules               |
+| $PSHOME modules     | C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules                       | c:\program files\powershell\7-preview\Modules     |
+| Windows modules     |                                                                          | C:\WINDOWS\system32\WindowsPowerShell\v1.0\Module |
+| App added modules   | C:\Program Files (x86)\Microsoft Azure Information Protection\Powershell |                                                   |
+
+Currently, applications (or user) adding additional paths to `$env:PSModulePath` are dropped in PowerShell 7.
+
+### Windows PowerShell PSModulePath construction
+
+Windows PowerShell has logic at startup to construct the PSModulePath based on (simplified version):
+
+- If PSModulePath doesn't exist: Combine `User` modules path, `System` modules path, and `$PSHOME` modules path
+- If PSModulePath does exist:
+  - If PSModulePath contains `$PSHOME` modules path:
+    - `System` modules path is inserted before `$PSHOME` modules path if it's not there
+  - else:
+    - Just use `$env:PSModulePath` as-is as user had deliberately removed `$PSHOME`
+
+`User` module path is prefixed only if User scope `$env:PSModulePath` doesn't exist.
+Otherwise, it is assumed the user removed it if it's not there already.
+
+### PowerShell 7 startup
+
+Currently, PS7 doesn't use contents of `$env:PSModulePath` if it detects it was started from PowerShell
+and simply overwrites it with `User` modules path + `System` modules path + `$PSHOME` modules path + `Windows` modules path.
+
+Change would be to use `$env:PSModulePath`, but prefix with `User` modules path + `System` modules path +
+`$PSHOME` modules path if any of those paths are not already there.
+
+`Windows` modules path will already be there along with any additional paths added by the user or applications.
+If the `Windows` modules path is not in `$env:PSModulePath`, then it will not be added as the user removed it.
+
+### Starting Windows PowerShell from PowerShell 7 Implementation
+
+Note that Windows PowerShell here means both `powershell.exe` and `powershell_ise.exe`.
+
+Copy process `$env:PSModulePath` as `WinPSModulePath`:
+
+- Remove PS7 `User` module path
+- Remove PS7 `System` module path
+- Remove PS7 `$PSHOME` module path
+
+Use that `WinPSModulePath` when starting Windows PowerShell.
+
+### Starting PowerShell 7 from Windows PowerShell
+
+The PowerShell 7 startup covers this case and should just work.
```

Comment:
these need to be explicit; what changes are made (if any) to `env:PSModulePath` in these cases? Are there differences between platforms (Windows vs Non-Windows)